sway/desktop/xwayland.c

```diff
@@ -274,7 +296,21 @@ static void set_activated(struct sway_view *view, bool activated) {
 	}
 
 	wlr_xwayland_surface_activate(surface, activated);
-	wlr_xwayland_surface_restack(surface, NULL, XCB_STACK_MODE_ABOVE);
+	if (activated) {
+		wlr_xwayland_surface_restack(surface, NULL, XCB_STACK_MODE_ABOVE);
+	}
+
+	struct sway_xwayland_unmanaged *unmanaged;
+	wl_list_for_each(unmanaged, &root->xwayland_unmanaged, link) {
+		struct wlr_xwayland_surface *unmanaged_surface = unmanaged->wlr_xwayland_surface;
+		if (!is_surface_transient_for(unmanaged_surface, view)) {
+			continue;
+		}
+		unmanaged->visible = activated;
+		if (activated) {
+			wlr_xwayland_surface_restack(unmanaged_surface, NULL, XCB_STACK_MODE_ABOVE);
```

Comment:
Alternatively we could ensure that no window ever gets restacked above anything that's override_redirect...